java/src/main/java/com/google/openlocationcode/OpenLocationCode.java

```diff
@@ -555,79 +556,89 @@ public String toString() {
    * @return True if it is a valid full or short code.
    */
   public static boolean isValidCode(String code) {
+    return toValidCode(code) != null;
+  }
+
+  /**
+   * Normalizes Open Location, or returns `null` if the code is not valid.
+   *
+   * @param code The code to check and normalize.
+   * @return Normalized code if it is a valid full or short code, null otherwise.
+   */
+  private static String toValidCode(String code) {
     if (code == null || code.length() < 2) {
-      return false;
+      return null;
     }
-    code = code.toUpperCase();
+    code = code.toUpperCase(Locale.ROOT);
 
     // There must be exactly one separator.
     int separatorPosition = code.indexOf(SEPARATOR);
     if (separatorPosition == -1) {
-      return false;
+      return null;
     }
     if (separatorPosition != code.lastIndexOf(SEPARATOR)) {
-      return false;
+      return null;
     }
     // There must be an even number of at most 8 characters before the separator.
     if (separatorPosition % 2 != 0 || separatorPosition > SEPARATOR_POSITION) {
-      return false;
+      return null;
     }
 
     // Check first two characters: only some values from the alphabet are permitted.
     if (separatorPosition == SEPARATOR_POSITION) {
       // First latitude character can only have first 9 values.
       if (CODE_ALPHABET.indexOf(code.charAt(0)) > 8) {
-        return false;
+        return null;
       }
 
       // First longitude character can only have first 18 values.
       if (CODE_ALPHABET.indexOf(code.charAt(1)) > 17) {
-        return false;
+        return null;
       }
     }
 
     // Check the characters before the separator.
-    boolean paddingStarted = false;
+    boolean paddingStarted = null;
```

Comment:
```suggestion
    boolean paddingStarted = false;
```

This breaks the tests ("incompatible types: \<null\> cannot be converted to boolean").